src/Compilers/CSharp/Portable/Emitter/Model/NamedTypeSymbolAdapter.cs

```diff
@@ -407,7 +407,6 @@ IEnumerable<Cci.IFieldDefinition> Cci.ITypeDefinition.GetFields(EmitContext cont
 
             foreach (var f in GetFieldsToEmit())
             {
-                Debug.Assert((object)(f.TupleUnderlyingField ?? f) == f);
```

Comment:
>Debug.Assert((object)(f.TupleUnderlyingField ?? f) == f); [](start = 16, length = 57)

I think we should keep the assert.

Reply:
In fact, I think we should make it even stronger and make sure we don't run into a ```TupleErrorFieldSymbol```.

---
In reply to: [426203595](https://github.com/dotnet/roslyn/pull/44231#discussion_r426203595) [](ancestors = 426203595)

Reply:
s

---
In reply to: [426264743](https://github.com/dotnet/roslyn/pull/44231#discussion_r426264743) [](ancestors = 426264743,426203595)